src/process/loader-shell-runner.ts

```diff
@@ -71,8 +73,20 @@ export class LoaderShellRunner implements IShellRunner {
     this.tty.showCursor();
   }
 
+  /**
+   * In case of an exception you probably want to force stop the loading animation. Oftherwise the parallel
+   * running commands would still keep the loader spinning which messes with the error text display.
+   * This just ignores the commands running in the background and kills off the animation right away.
+   */
+  private forceStopLoading() {
+    this.runningCommandsCount = 0;
+    this.stopLoading();
+  }
+
   private stopLoading() {
-    this.runningCommandsCount--;
+    if (this.runningCommandsCount > 0) {
+      this.runningCommandsCount--;
+    }
     if (this.runningCommandsCount > 0) {
       return;
     }
```

Comment:
meta: maybe the readability of this class can be slightly improved/refactored by separating the loader presentation from the loader logic/controller.

Reply:
I once worked upon it a bit, but its a bit chaotic since you need to have some wild jumps between the blocks. I guess for now its okaish as I hope this class does not need to be touched anymore. :D But we need to keep that in mind if it gets even more complicated we should rework its implementation.